File: apps/desktop/src/renderer/src/lib/api/endpoints/afm.ts
/** AFM endpoints — where each agent's content lives on disk. */
import { requestJson } from "../http";
import type { AfmApplyResult, AfmArchiveResult, AfmStatus } from "../types";

/** `GET /afm/status` — root + per-agent/per-team folder placement. */
export function getAfmStatus(): Promise<AfmStatus> {
  return requestJson<AfmStatus>("/afm/status", { timeoutMs: 40_000 });
}

/** `POST /afm/apply` — choose the root (empty → Evano default) and organize.
 *  Moves agent folders and may restart the gateway, hence the long timeout. */
export function applyAfm(root?: string | null): Promise<AfmApplyResult> {
  return requestJson<AfmApplyResult>("/afm/apply", {
    method: "POST",
    body: { root: root ?? null },
    timeoutMs: 180_000,
  });
}

/** `POST /afm/archive-team-run` — save a finished team run under Teams/<Team>/. */
export function archiveTeamRun(body: {
  team_name: string;
  steps: { agent_id: string; output: string }[];
  final: string;
}): Promise<AfmArchiveResult> {
  return requestJson<AfmArchiveResult>("/afm/archive-team-run", {
    method: "POST",
    body,
    timeoutMs: 60_000,
  });
}

File: services/agent-engine/app/api/afm.py
"""AFM endpoints — where each agent's content lives on disk."""

from __future__ import annotations

from fastapi import APIRouter, Depends

from ..schemas.afm import (
    AfmApplyRequest,
    AfmApplyResult,
    AfmArchiveRequest,
    AfmArchiveResult,
    AfmStatusResponse,
)
from ..services.afm_service import AFMService
from .deps import get_afm_service

router = APIRouter(prefix="/afm", tags=["afm"])


@router.get("/status", response_model=AfmStatusResponse, summary="AFM status")
def afm_status(service: AFMService = Depends(get_afm_service)) -> dict:
    """Current root, and per-agent/per-team folder placement."""
    return service.status()


@router.post("/apply", response_model=AfmApplyResult, summary="Choose the root and organize")
def afm_apply(
    payload: AfmApplyRequest,
    service: AFMService = Depends(get_afm_service),
) -> dict:
    """Point every agent's OpenClaw workspace at <root>/Agents/<Name> (moving the
    existing files), scaffold Teams folders, and reload the gateway."""
    return service.apply(payload.root)


@router.post("/archive-team-run", response_model=AfmArchiveResult, summary="Save a team run")
def afm_archive_team_run(
    payload: AfmArchiveRequest,
    service: AFMService = Depends(get_afm_service),
) -> dict:
    """Store each member's step output and the final result under Teams/<Team>/."""
    return service.archive_team_run(
        team_name=payload.team_name,
        steps=[s.model_dump() for s in payload.steps],
        final=payload.final,
    )
